internal/btf/btf.go

```diff
@@ -299,6 +295,20 @@ func parseBTFHeader(r io.Reader, bo binary.ByteOrder) (*btfHeader, error) {
 	return &header, nil
 }
 
+func guessRawBTFByteOrder(r io.ReaderAt) binary.ByteOrder {
+	for _, bo := range []binary.ByteOrder{
+		binary.LittleEndian,
+		binary.BigEndian,
+	} {
+		_, err := parseBTFHeader(internal.NewBufferedSectionReader(r, 0, math.MaxInt64), bo)
```

Comment:
I think the use of `NewBufferedSectionReader` is unnecessary, `io.NewSectionReader` should be sufficient.